packages/flutter_tools/test/commands.shard/permeable/create_test.dart

```diff
@@ -286,6 +286,66 @@ void main() {
     ),
   });
 
+  testUsingContext('creates a new project with flutter version constraint in pubspec', () async {
+    Cache.flutterRoot = '../..';
+
+    final CreateCommand command = CreateCommand();
+    final CommandRunner<void> runner = createTestCommandRunner(command);
+
+    await runner.run(<String>['create', '--no-pub', '--template=app', projectDir.path]);
```

Comment:
I modified all the pubspec templates to keep them consistent with this change.

However, I saw that some templates had their Flutter version pinned already.
Do we change those in this PR, or do we keep them unchanged?

In lieu of that change, I haven't yet modified the test to do the same expectation for all the template types. (although this can be easily be added)

Before I do that, I want to get some opinions from the tools team first :)